src/tasks/minions/farmingActivity.ts

```diff
@@ -0,0 +1,426 @@
+				if (plantToHarvest.variableYield) {
+					cropYield = calcVariableYield(
+						plantToHarvest,
+						patchType.lastUpgradeType,
+						currentFarmingLevel
+					);
+				} else if (plantToHarvest.fixedOutput) {
+					if (!plantToHarvest.fixedOutputAmount) return;
+					cropYield = plantToHarvest.fixedOutputAmount;
+				} else {
+					const plantChanceFactor =
+						Math.floor(
+							Math.floor(
+								plantToHarvest.chance1 +
+									(plantToHarvest.chance99 - plantToHarvest.chance1) *
+										((user.skillLevel(SkillsEnum.Farming) - 1) / 98)
+							) * baseBonus
+						) + 1;
+					const chanceToSaveLife = (plantChanceFactor + 1) / 256;
+					if (plantToHarvest.seedType === 'bush') lives = 4;
+					cropYield = 0;
+					const livesHolder = lives;
+					for (let k = 0; k < alivePlants; k++) {
+						lives = livesHolder;
+						for (let n = 0; lives > 0; n++) {
+							if (Math.random() > chanceToSaveLife) {
+								lives -= 1;
+								cropYield += 1;
+							} else {
+								cropYield += 1;
+							}
+						}
+					}
+				}
+
+				if (quantity > patchType.lastQuantity) {
+					loot[plantToHarvest.outputCrop] = cropYield;
+					loot[itemID('Weeds')] = quantity - patchType.lastQuantity;
+				} else {
+					loot[plantToHarvest.outputCrop] = cropYield;
+				}
+
+				if (plantToHarvest.name === 'Limpwurt') harvestXp = plantToHarvest.harvestXp;
+				else harvestXp = cropYield * plantToHarvest.harvestXp;
+			}
+
+			if (plantToHarvest.needsChopForHarvest) {
+				if (!plantToHarvest.treeWoodcuttingLevel) return;
+				if (currentWoodcuttingLevel >= plantToHarvest.treeWoodcuttingLevel) chopped = true;
+				else {
+					await user.settings.sync(true);
+					const GP = user.settings.get(UserSettings.GP);
+					if (GP < 200 * alivePlants) {
+						throw `You do not have the required woodcutting level or enough GP to clear your patches in order to be able to plant more.`;
+					} else {
+						payStr = `*You did not have the woodcutting level required, so you paid a nearby farmer 200 GP per patch to remove the previous tree.*`;
+						await user.removeGP(200 * alivePlants);
+					}
+
+					harvestXp = 0;
+				}
+				if (plantToHarvest.givesLogs && chopped) {
+					if (!plantToHarvest.outputLogs) return;
+					if (!plantToHarvest.woodcuttingXp) return;
+					const amountOfLogs = rand(5, 10);
+					loot[plantToHarvest.outputLogs] = amountOfLogs * alivePlants;
+
+					woodcuttingXp += alivePlants * amountOfLogs * plantToHarvest.woodcuttingXp;
+					wcStr = ` You also received ${woodcuttingXp.toLocaleString()} Woodcutting XP.`;
+
+					harvestXp = 0;
+				} else if (plantToHarvest.givesCrops && chopped) {
+					if (!plantToHarvest.outputCrop) return;
+					loot[plantToHarvest.outputCrop] = cropYield * alivePlants;
+
+					harvestXp = cropYield * alivePlants * plantToHarvest.harvestXp;
+				}
+			}
+
+			if (quantity > patchType.lastQuantity) {
+				loot[6055] = (quantity - patchType.lastQuantity) * 3; // weeds
```

Comment:
Never put comments on the end of a line like that.

If `itemID('Weeds')` doesnt work here (which is what is preferred), then define it in a variable like: `WEEDS_ID`.